### PR TITLE
Pass in the area name instead of the view name to a parameter that requires an area name.

### LINKED ISSUE #3556: Two controllers with the same name under different areas (rc2)
I have two controllers named `Component`, one residing in the usual root (/Controllers) and one under an area called `Admin`, residing at /Areas/Admin/Controllers.  It seems the correct controller is picked based on the URL and route as I visit them, but the view file (e.g. index.cshtml) picked is always the first one I visit in that session...not the correct view for the area.

1) visit /admin/component/index (works)
2) then visit /component/index in the same session (throws the error below)

If I restart kestrel, I can reverse the visit order and the pattern repeats, i.e. each page works in isolation.

```
fail: Microsoft.AspNet.Diagnostics.ExceptionHandlerMiddleware[0]
      An unhandled exception has occurred: Exception has been thrown by the target of an invocation.
      System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: The model item passed into the ViewDataDictionary is of type 'System.Collections.Generic.List`1[Kraken.Core.Layout.FileBootCard]', but this ViewDataDictionary instance requires a model item of type 'Kraken.Models.Component'.
         at Microsoft.AspNet.Mvc.ViewFeatures.ViewDataDictionary.EnsureCompatible (System.Object value) in <filename unknown>:line 0
         at Microsoft.AspNet.Mvc.ViewFeatures.ViewDataDictionary.SetModel (System.Object value) in <filename unknown>:line 0
         at Microsoft.AspNet.Mvc.ViewFeatures.ViewDataDictionary..ctor (Microsoft.AspNet.Mvc.ViewFeatures.ViewDataDictionary source, System.Object model, System.Type declaredModelType) in <filename unknow
```
and if I reverse the order I navigate to the pages:
```
fail: Microsoft.AspNet.Diagnostics.ExceptionHandlerMiddleware[0]
      An unhandled exception has occurred: Exception has been thrown by the target of an invocation.
      System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: The model item passed into the ViewDataDictionary is of type 'Kraken.Models.Component', but this ViewDataDictionary instance requires a model item of type 'System.Collections.Generic.IEnumerable`1[Kraken.Core.Layout.FileBootCard]'.
         at Microsoft.AspNet.Mvc.ViewFeatures.ViewDataDictionary.EnsureCompatible (System.Object value) in <filename unknown>:line 0
         at Microsoft.AspNet.Mvc.ViewFeatures.ViewDataDictionary.SetModel (System.Object value) in <filename unknown>:line 0
         at Microsoft.AspNet.Mvc.ViewFeatures.ViewDataDictionary..ctor (Microsoft.AspNet.Mvc.ViewFeatures.ViewDataDictionary source, System.Object model, System.Type declaredModelType) in <filename unknown>:line 0

```

DNX : dnx-mono.1.0.0-rc2-16177
MVC deps : 6.0.0-rc2-16377

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #3556

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @dougbu

**Comment by rynowak** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.Razor/RazorViewEngine.cs [hunk 1]
```diff
@@ -312,11 +312,11 @@ private ViewLocationCacheResult LocatePageFromPath(string pageName, bool isParti
...
-                expanderContext.ViewName,
-                expanderContext.ControllerName,
-                expanderContext.ViewName,
-                expanderContext.IsPartial,
-                expanderValues);
+                viewName: expanderContext.ViewName,
+                controllerName: expanderContext.ControllerName,
+                areaName: expanderContext.AreaName,
+                isPartial: expanderContext.IsPartial,
+                values: expanderValues);
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Razor/RazorViewEngine.cs]: Why bother naming the parameters? Would have helped when the code was originally written but looks overly verbose now.

**Comment by dougbu** [comment]
One minor comment then :shipit:

File: src/Microsoft.AspNet.Mvc.Razor/RazorViewEngine.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Razor/RazorViewEngine.cs]: It was to repent for my carelessness. Removed :+1:

**Pull request closed without merging**